x-pack/plugins/apm/public/components/app/service_groups/service_group_save/group_details.tsx

```diff
@@ -0,0 +1,208 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import {
+  EuiButton,
+  EuiButtonEmpty,
+  EuiModalBody,
+  EuiModalFooter,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiColorPicker,
+  EuiFieldText,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFormRow,
+  useColorPickerState,
+  EuiText,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import React, { useEffect, useRef, useState } from 'react';
+import type { StagedServiceGroup } from './save_modal';
+
+interface Props {
+  serviceGroup?: StagedServiceGroup;
+  isEdit?: boolean;
+  onCloseModal: () => void;
+  onClickNext: (serviceGroup: StagedServiceGroup) => void;
+  onDeleteGroup: () => void;
+  isLoading: boolean;
+}
+
+export function GroupDetails({
+  isEdit,
+  serviceGroup,
+  onCloseModal,
+  onClickNext,
+  onDeleteGroup,
+  isLoading,
+}: Props) {
+  const [name, setName] = useState<string>(serviceGroup?.groupName || '');
+  const [color, setColor, errors] = useColorPickerState(
```

Comment:
Perhaps make this more specific. 

```suggestion
  const [color, setColor, colorPickerErrors] = useColorPickerState(
```